File: python/high-scores/high_scores.py
class HighScores(object):
    def __init__(self, scores):
        self.scores = scores

    def latest(self):
        return self.scores[-1]

    def personal_best(self):
        return max(self.scores)

    def personal_top(self):
        return sorted(self.scores, reverse=True)[:3]

    def report(self):
        latest = self.latest()
        best = self.personal_best()
        diff = best - latest

        score = "Your latest score was {}.".format(latest)
        short = "That's {} short of your personal best!".format(diff)
        best = "That's your personal best!"

        return "{} {}".format(score, short if diff else best)

File: javascript/high-scores/high-scores.js
export class HighScores {
    constructor(scores) {
        this.scores = scores;

        this.latest = scores[scores.length - 1];

        this.highest = Math.max(...scores);

        this.top = [...scores].sort((a, b) => b - a).slice(0, 3);

        let score = `Your latest score was ${this.latest}.`;
        let personal = this.latest == this.highest
            ? `That's your personal best!`
            : `That's ${this.highest - this.latest} short of your personal best!`;
        this.report = `${score} ${personal}`;
    }
}
